Weather/ViewController.swift

```diff
@@ -8,12 +8,72 @@
 import UIKit
 
 class ViewController: UIViewController {
-
+    
+    @IBOutlet weak var TemperatureLabel: UILabel!
+    @IBOutlet weak var LocationLabel: UILabel!
+    @IBOutlet weak var searchTextField: UITextField!
+    
+    
+    // обявили переменную, указали что ее тип это протокол
+    let weatherService: WeatherServiceProtocol = WeatherService()
+    let locationService: LocationServiceProtocol = LocationService()
+    
+    
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
+        
+        Task {
+            guard let coordinates = await locationService.currentLocation() else { return }
+            await updateWeather(coordinates: coordinates)
+        }
+        
+        
+    }
+    
+    @IBAction func weatherCheckButtom(_ sender: UIButton) {
+        Task{
+            guard let city = searchTextField.text else { return }
+            
+            // полученый результат гарантировано не нил
+            guard let coordinades = await locationService.getLocation(name: city) else {
+                return
+            }
+            await updateWeather(coordinates: coordinades)
+            
+        }
+    }
+    
+    func updateWeather(coordinates: Coordinates) async {
+        guard let weather = await weatherService.weatherInLocation(location: coordinates) else {return}
+        let temperature = TemperatureLabel!
```

Comment:
We can ged rid of `temperature` and `location` local variables here, and use `LocationLabel` and `TemperatureLabel` properties directly.

Btw, you don't need to force-unwrap(with `!`) these properties, because it's already declared as force-unwrapped (`UILabel!`)